File: tests/python/ice-test/service/send_service.py
"""Mock send service for testing."""

import logging

logger = logging.getLogger(__name__)


class SendService:
    """Provides mock send operations."""
    
    def send_amount(self, uid: int, value: float) -> bool:
        """Simulate sending amount to a user."""
        logger.info(f"=======send amount uid={uid} value={value}")
        return True
    
    def send_point(self, uid: int, value: float) -> bool:
        """Simulate sending points to a user."""
        logger.info(f"=======send point uid={uid} value={value}")
        return True


# Global instance
send_service = SendService()

File: tests/go/ice-test/service/send_service.go
// Package service provides mock services for testing.
package service

import (
	"log/slog"
)

// SendService provides mock send operations.
type SendService struct{}

// NewSendService creates a new SendService.
func NewSendService() *SendService {
	return &SendService{}
}

// SendAmount simulates sending amount to a user.
func (s *SendService) SendAmount(uid int, value float64) bool {
	slog.Info("=======send amount", "uid", uid, "value", value)
	return true
}

// SendPoint simulates sending points to a user.
func (s *SendService) SendPoint(uid int, value float64) bool {
	slog.Info("=======send point", "uid", uid, "value", value)
	return true
}

